packages/editor/src/store/selectors.js

```diff
@@ -691,9 +693,16 @@ export const getBlocks = createSelector(
 			( clientId ) => getBlock( state, clientId )
 		);
 	},
-	( state ) => [
-		state.editor.present.blocks,
-	]
+	( state, rootClientId ) => {
+		if ( ! rootClientId ) {
+			return [
+				state.editor.present.blocks,
+			];
+		}
+		return [
+			getBlock.getDependants( state, rootClientId ),
```

Comment:
This is wrong because it will produce a nested array. `getDependants` returns a newly-generated array every time it's called, so the dependants will always be different. This should either (a) spread `...` into the return array or (b) be returned directly as the return value.

Reply:
Excellent catch! Silly mistake; fixed.